Updated iddarea.json, phoneloc.json and numplan.json from phonenumber library based on metadata version 4

gouniLee opened the pull request:
### Checklist

* [ ] At least one test case is included for this feature or bug fix.
* [x] `ReleaseNotes` has added or is not needed.
* [x] This PR has passed all of the test cases on `Nodejs` and `Chrome Browser`.
* [x] This PR has passed all of the test cases on `QT/QML`.
* [ ] This is an API breaking change.
* [ ] Requires a major version change.


### Issue Resolved / Feature Added
Updated `iddarea.json`, `phoneloc.json`, and `numplan.json` from phonenumber library based on metadata version 4.0.0 which are used in PhoneNumber formatting

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)

### Links
[//]: # (Related issues, references)

ehoogerbeets commented on docs/ReleaseNotes.md:
```diff
@@ -1,6 +1,15 @@
+-------
+Published as version 14.8.1
+
+New Features:
```
New features require a minor version update, so v14.9.0. But, I think this is not a new feature. It's more of an update/fix to existing features. I would put it in the bug fixes section instead

ehoogerbeets reviewed (commented):
No unit test updates are needed to verify these? Maybe add 3 or 4 of them?

gouniLee commented:
> No unit test updates are needed to verify these? Maybe add 3 or 4 of them?

I've added some test cases. and updated ReleaseNotes.
In addition, I've added a new sampleNumbers.json file that was used locally. The data is generated from libphonenumber's metadata file.  I think it would be useful to make test cases as the reference numbers.
